src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_COUNT = 4;
+    private static final int MAX_RADIUS = 10;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(MAX_COUNT);
+        switch (index) {
+            case 0:
+                return new Circle(new Random().nextInt(), colorSupplier.getRandomColor());
```

Comment:
move to private method
```suggestion
                return new prepareCircle();
```